### PR TITLE
Use recommended dictConfig for logging configuration

### PR DESCRIPTION
**Pull request by glatterf42**
This PR is in tandem with https://github.com/iiasa/ixmp4/pull/83, where you can find an extensive description of the underlying issue. For pyam, there is no `server.conf`, so these parts can be disregarded.

Does this PR need to be mentioned in the release notes?
Also, can we please have a quick patch release after merging this?

### CODE REVIEW

File: pyam/logging.py [hunk 1]
```diff
@@ -11,7 +12,10 @@
 
 def configure_logging():
     """Configure logging"""
-    config.fileConfig(here / "logging.conf", disable_existing_loggers=False)
+    logging_config = here / "logging.json"
+    with open(logging_config, "r") as file:
```
Comment (danielhuppmann) [inline pyam/logging.py]:
```suggestion
    with open(logging_config) as file:
```

File: pyam/logging.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      }
+    },
+    "handlers": {
+      "console": {
+        "class": "logging.StreamHandler",
+        "level": "NOTSET",
+        "formatter": "generic"
+      }
+    }
+  }
```
Comment (danielhuppmann) [inline pyam/logging.json]:
```suggestion
  }

```

File: pyam/logging.py [hunk 3]
```diff
@@ -11,7 +12,10 @@
...
+        config_dict = json.load(file)
+    config.dictConfig(config_dict)
```
Comment (danielhuppmann) [inline pyam/logging.py:18]:
```suggestion
        config.dictConfig(json.load(file))
```

**Review by danielhuppmann — APPROVED** [review]
Thanks, a few minor suggestions inline...

File: pyam/logging.py (hunk 3 above)
Comment (glatterf42) [reply pyam/logging.py:18]: Ah, one more than anticipated ;)

**Review by danielhuppmann — APPROVED** [review]
Thanks @glatterf42, hope that this will resolve some downstream issues...

**Pull request merged**